src/core/border_router/routing_manager.cpp

```diff
@@ -162,25 +164,53 @@ Error RoutingManager::GetNat64Prefix(Ip6::Prefix &aPrefix)
 }
 #endif
 
-Error RoutingManager::LoadOrGenerateRandomOmrPrefix(void)
+Error RoutingManager::LoadOrGenerateRandomBrUlaPrefix(void)
 {
     Error error = kErrorNone;
 
-    if (Get<Settings>().Read<Settings::OmrPrefix>(mLocalOmrPrefix) != kErrorNone || !IsValidOmrPrefix(mLocalOmrPrefix))
+    if (Get<Settings>().Read<Settings::BrUlaPrefix>(mBrUlaPrefix) != kErrorNone || !IsValidBrUlaPrefix(mBrUlaPrefix))
     {
-        Ip6::NetworkPrefix randomOmrPrefix;
+        Ip6::NetworkPrefix randomBrUlaPrefix;
 
-        LogNote("No valid OMR prefix found in settings, generating new one");
+        LogNote("No valid /48 BR ULA prefix found in settings, generating new one");
 
-        // TODO: generate OMR prefix from the /48 BR ULA prefix
-        error = randomOmrPrefix.GenerateRandomUla();
+        error = randomBrUlaPrefix.GenerateRandomUla();
         if (error != kErrorNone)
         {
-            LogCrit("Failed to generate random OMR prefix");
+            LogCrit("Failed to generate random /48 BR ULA prefix");
             ExitNow();
         }
 
-        mLocalOmrPrefix.Set(randomOmrPrefix);
+        randomBrUlaPrefix.m8[6] = 0;
+        randomBrUlaPrefix.m8[7] = 0;
+        mBrUlaPrefix.Set(randomBrUlaPrefix);
+        mBrUlaPrefix.SetLength(kBrUlaPrefixLength);
+
+        IgnoreError(Get<Settings>().Save<Settings::BrUlaPrefix>(mBrUlaPrefix));
+    }
+
+exit:
+    return error;
+}
+
+Error RoutingManager::LoadOrGenerateRandomOmrPrefix(void)
+{
+    Error error = kErrorNone;
+
+    if (Get<Settings>().Read<Settings::OmrPrefix>(mLocalOmrPrefix) != kErrorNone || !IsValidOmrPrefix(mLocalOmrPrefix))
```

Comment:
Do we still save the whole OMR prefix in settings? 
Or can we just save the 16 bit network ID for OMR, which builds the final OMR prefix with /48  BR ULA prefix?

Reply:
I found that if the network ID of OMR prefix is fixed, we don't need to save OMR prefix to settings at all. 
We only need to save the BR /48 ULA prefix.
Thoughts? @xuyirio

Reply:
Updated to only save the /48 BR ULA prefix.